src/Agda2Hs/Compile/Utils.hs

```diff
@@ -165,15 +165,24 @@ isClassModule m
   | null $ mnameToList m = return False
   | otherwise            = do
     minRec <- asks minRecordName
-    getConstInfo' (mnameToQName m) >>= \case
-      _ | Just m == minRec -> return True
-      Right Defn{defName = r, theDef = Record{}} ->
-        -- It would be nicer if we remembered this from when we looked at the record the first time.
-        processPragma r <&> \case
-          ClassPragma _       -> True
-          ExistingClassPragma -> True
-          _                   -> False
-      _                       -> return False
+    if Just m == minRec then return True else isClassName (mnameToQName m)
+
+isClassName :: QName -> C Bool
```

Comment:
The name kinda gives it away, but can you please add a Hoogle comment?

Reply:
We don't really have Haddock comments for *any* of the functions, so just adding the one here feels like it would be a bit out of place.